slides/intro-extra-tune.qmd

```diff
@@ -0,0 +1,132 @@
+---
+title: "Extras - Tuning"
+subtitle: "Introduction to tidymodels"
+format:
+  revealjs: 
+    slide-number: true
+    footer: <https://workshops.tidymodels.org>
+    include-before-body: header.html
+    include-after-body: footer-annotations.html
+    theme: [default, tidymodels.scss]
+    width: 1280
+    height: 720
+knitr:
+  opts_chunk: 
+    echo: true
+    collapse: true
+    comment: "#>"
+    fig.path: "figures/"
+---
+
+```{r}
+#| label: setup
+#| include: false
+#| file: setup.R
+```
+
+```{r setup-previous}
+#| echo: false
+library(tidymodels)
+library(modeldatatoo)
+
+taxi <- data_taxi()
+
+set.seed(123)
+taxi_split <- initial_split(taxi, prop = 0.8, strata = tip)
+taxi_train <- training(taxi_split)
+taxi_test <- testing(taxi_split)
+
+set.seed(123)
+taxi_folds <- vfold_cv(taxi_train, v = 10, strata = tip)
+```
+
+## Tuning parameters
+
+Some model or preprocessing parameters cannot be estimated directly from the data.
+
+. . .
+
+Some examples:
+
+- Tree depth in decision trees
+- Number of neighbors in a K-nearest neighbor model
+
+## Optimize tuning parameters
+
+- Try different values and measure their performance.
+
+. . .
+
+- Find good values for these parameters.
+
+. . .
+
+- Once the value(s) of the parameter(s) are determined, a model can be finalized by fitting the model to the entire training set.
+
+## Optimize tuning parameters
+
+The main two strategies for optimization are:
+
+. . .
+
+-   **Grid search** 💠 which tests a pre-defined set of candidate values
+
+-   **Iterative search** 🌀 which suggests/estimates new values of candidate parameters to evaluate
+
+## Specifying tuning parameters
+
+Let's take our previous random forest workflow and tag the number of trees for tuning:
+
+```{r}
+#| label: tag-for-tuning
+#| code-line-numbers: "1|"
+
+rf_spec <- rand_forest(trees = tune()) %>% 
```

Comment:
This is definitely the most intuitive model argument to `rand_forest()`, but I'm a bit [hesitant](https://stats.stackexchange.com/questions/348245/do-we-have-to-tune-the-number-of-trees-in-a-random-forest) to demonstrate tuning with it. It also looks like it doesn't result in marked changes in performance in this example. Would we be down for tuning `mtry` or `min_n` instead?

Reply:
`mtry()` would/could mean talking about finalizing parameters so rather `min_n`?

Reply:
Totally!

Reply:
done! [42b756b](https://github.com/tidymodels/workshops/pull/239/commits/42b756b33190be2eb6da6551d1a2ef7b0ed4d819)